File: ios/SwiftKotlination/SwiftKotlinationTests/Sources/Features/Story/Factory/Mocks/StoryFactoryMock.swift
@testable import SwiftKotlination

struct StoryFactoryMock {
    let storyManager: StoryManagerProtocol
    let imageManager: ImageManagerMock

    init(storyManager: StoryManagerProtocol = StoryManagerMock(), imageManager: ImageManagerMock = ImageManagerMock()) {
        self.storyManager = storyManager
        self.imageManager = imageManager
    }
}

extension StoryFactoryMock: StoryFactory {
    func makeStoryManager(for story: Story) -> StoryManagerProtocol {
        return storyManager
    }

    func makeStoryViewController(for story: Story) -> StoryViewController {
        let viewController = StoryViewController.storyBoardInstance
        viewController.viewModel = StoryViewModel(factory: self, story: story)

        return viewController
    }
}

extension StoryFactoryMock: ImageFactory {
    func makeImageManager() -> ImageManagerProtocol {
        return imageManager
    }
}

File: android/SwiftKotlination/app/src/test/kotlin/fr/jhandguy/swiftkotlination/features/story/factory/mocks/StoryFactoryMock.kt
package fr.jhandguy.swiftkotlination.features.story.factory.mocks

import android.app.Activity
import fr.jhandguy.swiftkotlination.coordinator.Coordinator
import fr.jhandguy.swiftkotlination.coordinator.CoordinatorInterface
import fr.jhandguy.swiftkotlination.features.story.factory.StoryFactory
import fr.jhandguy.swiftkotlination.features.story.model.Story
import fr.jhandguy.swiftkotlination.features.story.model.StoryManagerInterface
import fr.jhandguy.swiftkotlination.features.story.viewModel.StoryViewModel
import fr.jhandguy.swiftkotlination.model.ImageManagerInterface

class StoryFactoryMock(val storyManager: StoryManagerInterface, val imageManager: ImageManagerInterface) : StoryFactory {
    override fun makeStoryManager(story: Story): StoryManagerInterface = storyManager
    override fun makeStoryViewModel(story: Story): StoryViewModel = StoryViewModel(this, story)
    override fun makeCoordinator(activity: Activity): CoordinatorInterface = Coordinator(activity)
    override fun makeImageManager(): ImageManagerInterface = imageManager
}